### PR TITLE
DoIts without method header and reformatting

### PR DESCRIPTION
**Pull request by dionisiydk**
DoIts without method header and reformatting:
simpleDoIt

- RBParser is refactored to simplify method and doIts parsing/compilation
 - common parts are extracted to parseMethodBodyInto:
    - pragmas are supported in doIts (see #testPragmaInExpression):
      - it was some strange constraint

      - compilation flags can be now enabled in workspace
  - RBReturnNode is improved with better var name for position: #start instead of #return 
  - RBDoItMethodNode is introduced to represent method nodes in doIt ASTs
 - SHRBTextStyler>>isWorkspace uses isScripting which respects semantic scope to correctly highlight DoIts without headers

PR does not fix highlighting in the debugger and inspector for no-header DoIts. These fixes are in their repos:
- https://github.com/pharo-spec/NewTools/pull/400
- https://github.com/pharo-spec/Spec/pull/1296

### CODE REVIEW

**Comment by MarcusDenker** [comment]
Only failing test is 

- SpOptionListPresenterTest>>testDoAcceptIsNotExecutedWhenCancelingDialog

Which is not related to the change

**Comment by MarcusDenker** [comment]
Nice speedup for #evaluate:

```
[ Smalltalk compiler evaluate: '1+2' ] bench

before:  "'4431.741 per second'"
after:  "'28202.360 per second'"
```

File: src/AST-Core/ASTCacheMissStrategy.class.st [hunk 1]
```diff
@@ -16,5 +16,5 @@ Class {
 
 { #category : #accessing }
 ASTCacheMissStrategy >> getASTFor: aCompiledMethod [
-	^ aCompiledMethod parseTree doSemanticAnalysisIn: aCompiledMethod methodClass
+	^ aCompiledMethod parseTree
```
Comment (dionisiydk) [inline src/AST-Core/ASTCacheMissStrategy.class.st:19]: With little fix in #parseTree method (see below) the extra semantic analysis is not needed any more.

File: src/DrTests-CommentsToTests-Tests/CommentsToTestsTest.class.st [hunk 2]
```diff
@@ -28,7 +28,7 @@ CommentsToTestsTest >> testCommentWithSyntaxError [
 	docComment := thisContext method ast pharoDocCommentNodes first.
 	commentTestCase := CommentTestCase for: docComment.
 	
-	self should: [commentTestCase testIt] raise: SyntaxErrorNotification
+	self should: [commentTestCase testIt] raise: RuntimeSyntaxError
```
Comment (dionisiydk) [inline src/DrTests-CommentsToTests-Tests/CommentsToTestsTest.class.st:31]: This difference is from the previous reformatting logic of ast methods. There were two stages:
1. initial parsing of source code in faulty mode
2. reformatting and parsing again with normal mode which causes syntax notification.

Now the second step is removed and therefore only runtime syntax error can be thrown.

File: src/OpalCompiler-Core/OpalCompiler.class.st [hunk 3]
```diff
@@ -259,17 +259,7 @@ OpalCompiler >> compile: textOrString [
 { #category : #private }
 OpalCompiler >> compileDoItFromAST [
```
Comment (dionisiydk) [inline src/OpalCompiler-Core/OpalCompiler.class.st:260]: This is the main change to remove the header and reformatting

**Comment by MarcusDenker** [comment]
- ClassHierarchyPrinterTest>>testOnlyRBASTNodes needs to be update for the DoItMethodNode

File: src/DrTests-CommentsToTests-Tests/CommentsToTestsTest.class.st (hunk 2 above)
Comment (MarcusDenker) [reply src/DrTests-CommentsToTests-Tests/CommentsToTestsTest.class.st:31]: I always wanted to change the doc comments to store the AST, not the source, but that is for another PR of course

**Pull request merged**